src/fu-tool.c

```diff
@@ -3932,9 +3935,149 @@ fu_util_reboot_cleanup(FuUtilPrivate *priv, gchar **values, GError **error)
+		g_autofree gchar *partition_signature = fwupd_guid_to_string(
+		    fu_efi_hard_drive_device_path_get_partition_signature(dp_hdd),
+		    FWUPD_GUID_FLAG_MIXED_ENDIAN);
+
+		fwupd_codec_string_append(str, indent, "PartitionSignature", partition_signature);
+		fwupd_codec_string_append_hex(
+		    str,
+		    indent,
+		    "PartitionNumber",
+		    fu_efi_hard_drive_device_path_get_partition_number(dp_hdd));
+		fwupd_codec_string_append_hex(
+		    str,
+		    indent,
+		    "PartitionSize",
+		    fu_efi_hard_drive_device_path_get_partition_size(dp_hdd));
+		fwupd_codec_string_append_hex(
+		    str,
+		    indent,
+		    "PartitionOffset",
+		    fu_efi_hard_drive_device_path_get_partition_start(dp_hdd));
+
+		/* can we match the volume? */
+		volume =
+		    fu_context_get_esp_volume_by_hard_drive_device_path(priv->ctx, dp_hdd, error);
+		if (volume != NULL) {
+			fwupd_codec_string_append(str,
+						  indent,
+						  "MountPoint",
+						  fu_volume_get_mount_point(volume));
+		}
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_util_efivar_files(FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) files = NULL;
+
+	files = fu_context_get_esp_files(priv->ctx,
+					 FU_CONTEXT_ESP_FILE_FLAG_INCLUDE_FIRST_STAGE |
+					     FU_CONTEXT_ESP_FILE_FLAG_INCLUDE_SECOND_STAGE,
+					 error);
+	if (files == NULL)
+		return FALSE;
+	for (guint i = 0; i < files->len; i++) {
+		FuFirmware *firmware = g_ptr_array_index(files, i);
+		g_autofree gchar *name =
+		    g_strdup_printf("Boot%04X", (guint)fu_firmware_get_idx(firmware));
+		fu_console_print(priv->console,
+				 "%s → %s",
+				 name,
+				 fu_firmware_get_filename(firmware));
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_util_efivar_boot(FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	FuEfivars *efivars = fu_context_get_efivars(priv->ctx);
+	g_autoptr(GPtrArray) entries = NULL;
+	g_autoptr(GString) str = g_string_new(NULL);
+	guint16 idx = 0;
+
+	if (!fu_efivars_get_boot_current(efivars, &idx, error))
+		return FALSE;
+	fwupd_codec_string_append_hex(str, 0, "BootCurrent", idx);
+	if (fu_efivars_get_boot_next(efivars, &idx, NULL))
+		fwupd_codec_string_append_hex(str, 0, "BootNext", idx);
+
+	entries = fu_efivars_get_boot_entries(efivars, error);
+	if (entries == NULL)
+		return FALSE;
+	for (guint i = 0; i < entries->len; i++) {
+		FuEfiLoadOption *entry = g_ptr_array_index(entries, i);
+		g_autofree gchar *title =
+		    g_strdup_printf("Boot%04X", (guint)fu_firmware_get_idx(FU_FIRMWARE(entry)));
+		fwupd_codec_string_append(str, 0, title, "");
+		if (!fu_util_efivar_boot_entry(priv, entry, str, 1, error))
+			return FALSE;
+	}
+
+	/* success */
+	fu_console_print_literal(priv->console, str->str);
+	return TRUE;
+}
+
 static gboolean
 fu_util_efivar_list(FuUtilPrivate *priv, gchar **values, GError **error)
 {
+	FuEfivars *efivars = fu_context_get_efivars(priv->ctx);
 	g_autoptr(GPtrArray) names = NULL;
 
 	/* sanity check */
```

Comment:
it's not new code per-say, but I think it's logical to give a prompt here now so that even if you don't know what GUIDs you want you can see what is on your system.

Reply:
Yes, but lets do this in a different PR :)